In [ ]:
import pandas as pd
import numpy as np
print('Pandas is working perfectly fine')

Pandas is working perfectly fine


In [ ]:
# I use this code to read the csv file and display the first 5 rows of the dataset to notice any immmediate issues with the data. I also use the str.title() method to capitalize the first letter of each word in the 'Name' column for better readability.
df = pd.read_csv(r"C:\Users\Jasmine Agbahovbe\Downloads\MY CSV FILES\healthcare_dataset - Copy.csv")
df.head(5)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal


In [ ]:
#Then I use the str.title() method to capitalize the first letter of each word in the 'Name' column for better readability.
df['Name']=  df['Name'].str.title()
df.head(5)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby Jackson,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,Leslie Terry,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,Danny Smith,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,Andrew Watts,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,Adrienne Bell,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal


In [ ]:
df= df[df['Test Results']!= 'Inconclusive']
df.head(5)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby Jackson,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
2,Danny Smith,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,Andrew Watts,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,Adrienne Bell,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal
5,Emily Johnson,36,Male,A+,Asthma,2023-12-20,Taylor Newton,Nunez-Humphrey,UnitedHealthcare,48145.110951,389,Urgent,2023-12-24,Ibuprofen,Normal


In [ ]:
df['Medical Condition'].value_counts()
# I noticed that diabetes is the most common medical condition, so I checked the mean billing amount for patients in descending order
df.groupby('Medical Condition')['Billing Amount'].mean().sort_values(ascending=False)
# I then created a new dataframe to display the mean billing amount for each medical condition in descending order, as well as the frequency of each condition via joining.
mean_count = pd.DataFrame(df.groupby('Medical Condition')['Billing Amount'].mean().sort_values(ascending=False)).rename(columns={'Billing Amount':'Mean Billing Amount'}).join(df['Medical Condition'].value_counts()).rename(columns={'count':'Number of Patients'})
print('Average Billing Amount and Number of Patients for each Medical Condition:')
mean_count


Average Billing Amount and Number of Patients for each Medical Condition:


,Mean Billing Amount,Number of Patients
Medical Condition,,
Obesity,25825.923227,6189
Arthritis,25593.148558,6220
Diabetes,25548.482475,6258
Asthma,25534.237883,6156
Hypertension,25313.345451,6154
Cancer,25167.524747,6167


In [ ]:
#To look for correlation between blood type and medical condition, I created a new dataframe to display the mean billing amount for each medical condition in descending order, as well as the frequency of each condition via joining.
blood_count=pd.DataFrame(mean_count.join(df.groupby('Medical Condition')['Blood Type'].value_counts().rename('Blood Type Count'), how='right'))
blood_count

Mean Billing Amount  Number of Patients  \
Medical Condition Blood Type                                            
Arthritis         AB-                25593.148558                6220   
                  O+                 25593.148558                6220   
                  B-                 25593.148558                6220   
                  B+                 25593.148558                6220   
                  O-                 25593.148558                6220   
                  A-                 25593.148558                6220   
                  AB+                25593.148558                6220   
                  A+                 25593.148558                6220   
Asthma            AB+                25534.237883                6156   
                  A-                 25534.237883                6156   
                  O+                 25534.237883                6156   
                  O-                 25534.237883                6156   
                  AB-                25534.237883                6156   
                  B-                 25534.237883                6156   
                  A+                 25534.237883                6156   
                  B+                 25534.237883                6156   
Cancer            AB-                25167.524747                6167   
                  A+                 25167.524747                6167   
                  B+                 25167.524747                6167   
                  O-                 25167.524747                6167   
                  B-                 25167.524747                6167   
                  A-                 25167.524747                6167   
                  O+                 25167.524747                6167   
                  AB+                25167.524747                6167   
Diabetes          A-                 25548.482475                6258   
                  A+                 25548.482475                6258   
                  O+                 25548.482475                6258   
                  B+                 25548.482475                6258   
                  B-                 25548.482475                6258   
                  AB+                25548.482475                6258   
                  AB-                25548.482475                6258   
                  O-                 25548.482475                6258   
Hypertension      AB+                25313.345451                6154   
                  B-                 25313.345451                6154   
                  A-                 25313.345451                6154   
                  O+                 25313.345451                6154   
                  O-                 25313.345451                6154   
                  A+                 25313.345451                6154   
                  AB-                25313.345451                6154   
                  B+                 25313.345451                6154   
Obesity           B-                 25825.923227                6189   
                  AB-                25825.923227                6189   
                  O-                 25825.923227                6189   
                  A+                 25825.923227                6189   
                  AB+                25825.923227                6189   
                  B+                 25825.923227                6189   
                  A-                 25825.923227                6189   
                  O+                 25825.923227                6189   

                              Blood Type Count  
Medical Condition Blood Type                    
Arthritis         AB-                      830  
                  O+                       794  
                  B-                       794  
                  B+                       767  
                  O-                       766  
                  A-                

In [ ]:
#I wanted to remove the Number of Patients and the Mean Billing Amount column from the blood_count dataFrame.
blood_count=blood_count.drop(columns=['Mean Billing Amount']).drop(columns=['Number of Patients'])
blood_count

Blood Type Count
Medical Condition Blood Type                  
Arthritis         AB-                      830
                  O+                       794
                  B-                       794
                  B+                       767
                  O-                       766
                  A-                       763
                  AB+                      760
                  A+                       746
Asthma            AB+                      809
                  A-                       808
                  O+                       790
                  O-                       773
                  AB-                      762
                  B-                       744
                  A+                       735
                  B+                       735
Cancer            AB-                      830
                  A+                       809
                  B+                       800
                  O-                       782
                  B-                       769
                  A-                       739
                  O+                       736
                  AB+                      702
Diabetes          A-                       817
                  A+                       809
                  O+                       806
                  B+                       792
                  B-                       784
                  AB+                      761
                  AB-                      750
                  O-                       739
Hypertension      AB+                      823
                  B-                       790
                  A-                       786
                  O+                       775
                  O-                       772
                  A+                       748
                  AB-                      733
                  B+                       727
Obesity           B-                       795
                  AB-                      790
                  O-                       788
                  A+                       779
                  AB+                      767
                  B+                       758
                  A-                       758
                  O+                       754

Hypothesis Testing: Analysing Blood Type against Medical Condition via Chi-Square
Now that I created blood_count, I will use the Chi-Squared test I learnt from school via the scipy library.
This is my hypotheses:
H0 : There is no correlation between the blood type of the patient and their illness.(They are independent)
H1: There is a correlation between the blood type of the patient and their illness.(They are dependent)
I will be using a 1% significance level, though I will compare with other significance levels later on.

In [ ]:
# Had to sort a ModuleNotFoundError with scipy.stats in bash , but I managed to run the table!
import pandas as pd
from scipy.stats import chi2_contingency

contingency_table = pd.crosstab(df['Medical Condition'], df['Blood Type'])
print("Contingency Table between Medical Condition and Blood Type:")
print(contingency_table)

Contingency Table between Medical Condition and Blood Type:
Blood Type          A+   A-  AB+  AB-   B+   B-   O+   O-
Medical Condition                                        
Arthritis          746  763  760  830  767  794  794  766
Asthma             735  808  809  762  735  744  790  773
Cancer             809  739  702  830  800  769  736  782
Diabetes           809  817  761  750  792  784  806  739
Hypertension       748  786  823  733  727  790  775  772
Obesity            779  758  767  790  758  795  754  788


In [ ]:
# Generated the required statistical values for the contingency table using the chi2_contingency function from scipy.stats.
chi2_stat, p_value, dof, expected = chi2_contingency(contingency_table)
print("\nStatistical Values:") #Learnt about newline character \n.
print(f"Chi-Squared Statistic: {chi2_stat:.3f}") # And learnt about 'f' to print variables in a string, and the :._f to round to _ decimal places.
print(f"P-Value: {p_value:.4f}")


Statistical Values:
Chi-Squared Statistic: 48.951
P-Value: 0.0589


Conclusion:
With a P-Value of 0.0589, or approximately 5.9%, it is above the 1% sig. level and is still non-significant at a 5% significance level. 
Thus, I must accept H0: There is indeed independence between the blood type and the patient's likelihood of getting a certain disease in this dataset.

In [ ]:
avg_age=pd.DataFrame(df.groupby('Medical Condition')['Age'].mean().sort_values(ascending=False)).rename(columns={'Age':'Average Age'}).dropna()
avg_age['Average Age'] = avg_age['Average Age'].round(2) # Learnt about the round() function 
avg_age
# I realised that the average age of patients with medical conditions is roughly the same, at 51-52 years old.


Normal Distribution of Age
I was curious to see how the age of patients would be distributed, since the mean is around 51-52. So, I studied the norm functions in the scipy library to form a normal distribution curve, then used matplotlib along with some np functions I already know to plot on those values to form a static image. Moreover, I wanted to show the mean and standard deviation as stand alone statistics, so I used the f...{._f} step I did in the chi-squared test.


In [2]:
import matplotlib.pyplot as plt 
import numpy as np
from scipy.stats import norm 
%matplotlib inline
mu = df['Age'].mean()
sigma = df['Age'].std()
plt.hist(df['Age'], bins=30, density=True, alpha=0.5, color='blue', edgecolor='white')
xmin, xmax = plt.xlim()
x=np.linspace(xmin, xmax, 100)
p= norm.pdf(x,mu, sigma)

plt.plot(x,p,color='darkblue', linewidth=2.5,label=f'Normal curve\n(Mean={mu:2f},Standard Deviation={sigma:2f})')
plt.title('The Normal Curve Distribution of Age',fontsize=14,fontweight='bold')
plt.xlabel('Age', fontsize=12)
plt.ylabel('Density', fontsize=12)
plt.grid(axis='y', linestyle='--',alpha=0.3)
plt.legend(frameon=True, facecolor='white', edgecolor="none")
plt.show()

ModuleNotFoundError: No module named 'scipy'